symphony-bdk-http/symphony-bdk-http-webclient/src/main/java/com/symphony/bdk/http/webclient/ApiClientWebClient.java

```diff
@@ -0,0 +1,287 @@
+  public String getBasePath() {
+    return this.basePath;
+  }
+
+  @Override
+  public String parameterToString(Object param) {
+    if (param == null) {
+      return "";
+    } else if (param instanceof Collection) {
+      StringBuilder b = new StringBuilder();
+      for (Object o : (Collection<?>) param) {
+        if (b.length() > 0) {
+          b.append(',');
+        }
+        b.append(o);
+      }
+      return b.toString();
+    } else {
+      return String.valueOf(param);
+    }
+  }
+
+  @Override
+  public List<Pair> parameterToPairs(String collectionFormat, String name, Object value) {
+    List<Pair> params = new ArrayList<Pair>();
+
+    // preconditions
+    if (name == null || name.isEmpty() || value == null) {
+      return params;
+    }
+
+    Collection<?> valueCollection;
+    if (value instanceof Collection) {
+      valueCollection = (Collection) value;
+    } else {
+      params.add(new Pair(name, parameterToString(value)));
+      return params;
+    }
+
+    if (valueCollection.isEmpty()) {
+      return params;
+    }
+
+    // get the collection format (default: csv)
+    String format = (collectionFormat == null || collectionFormat.isEmpty() ? "csv" : collectionFormat);
+
+    // create the params based on the collection format
+    if ("multi".equals(format)) {
+      for (Object item : valueCollection) {
+        params.add(new Pair(name, parameterToString(item)));
+      }
+
+      return params;
+    }
+
+    String delimiter = ",";
+
+    switch (format) {
+      case "csv":
+        delimiter = ",";
+        break;
+      case "ssv":
+        delimiter = " ";
+        break;
+      case "tsv":
+        delimiter = "\t";
+        break;
+      case "pipes":
+        delimiter = "|";
+        break;
+    }
+
+    StringBuilder sb = new StringBuilder();
+    for (Object item : valueCollection) {
+      sb.append(delimiter);
+      sb.append(parameterToString(item));
+    }
+
+    params.add(new Pair(name, sb.substring(1)));
+
+    return params;
+  }
+
+  @Override
+  public String selectHeaderAccept(String[] accepts) {
+    if (accepts.length == 0) {
+      return null;
+    }
+    for (String accept : accepts) {
+      if (isJsonMime(accept)) {
+        return accept;
+      }
+    }
+    return String.join(",", accepts);
+  }
+
+  protected boolean isJsonMime(String mime) {
+    String jsonMime = "(?i)^(application/json|[^;/ \t]+/[^;/ \t]+[+]json)[ \t]*(;.*)?$";
+    return mime != null && (mime.matches(jsonMime) || mime.equals("*/*"));
+  }
+
+  @Override
+  public String selectHeaderContentType(String[] contentTypes) {
+    if (contentTypes.length == 0) {
+      return "application/json";
+    }
+    for (String contentType : contentTypes) {
+      if (isJsonMime(contentType)) {
+        return contentType;
+      }
+    }
+    return contentTypes[0];
+  }
+
+  @Override
+  public String escapeString(String str) {
+    try {
+      return URLEncoder.encode(str, "utf8").replaceAll("\\+", "%20");
```

Comment:
is that all that needs to be escaped? maybe there is something in spring already doing that?

Reply:
I'm not sure about this one. I just took the logic from the implementation of jersey2